tests/ssh/test_ssh_default_password.py

```diff
@@ -0,0 +1,50 @@
+import pytest
+import paramiko
+import logging
+
+pytestmark = [
+    pytest.mark.disable_loganalyzer,
+    pytest.mark.topology("any")
+]
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_USERNAME = "admin"
+DEFAULT_PASSWORD_PUBLIC = "YourPaSsWoRd"
+DEFAULT_PASSWORD_NON_PUBLIC = "password"
+
+
+def test_ssh_default_password(duthost):
+    """verify the initial SSH password is always expected.
+
+    When connecting SONiC via SSH, there are two kind of passwords:
+    1. if SONiC was built from public image, SSH password is YourPaSsWoRd.
+    2. if SONiC was built from non-public image, SSH password is password.
+
+    Args:
+        duthost: AnsibleHost instance for DUT
+    """
+
+    # set default password with public value
+    default_password = DEFAULT_PASSWORD_PUBLIC
+
+    # 1. check SONiC version
+    # If SONiC was built from non-public image, there should be information which has the key word 'sonic-dri' in motd
+    res = duthost.shell("cat /etc/motd | grep sonic-dri",module_ignore_errors=True)["stdout"]
+
+    if res:
+        default_password = DEFAULT_PASSWORD_NON_PUBLIC
```

Comment:
Probably it is a good idea to use a separate function to check the SONiC image type (public/microsoft/ or any other type).

Then each vendor can customize this function for their own needs.

Reply:
Good point, thanks. I will improve the code later.